## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 422218`
Commit: `935dc291c63252a3d8d8b6cd554fd93c711230d1`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>btop</li>
    <li>btop-cuda</li>
    <li>btop-rocm</li>
    <li>hyprpanel</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>btop</li>
    <li>btop-cuda</li>
    <li>hyprpanel</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>btop</li>
    <li>btop-cuda</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>btop</li>
    <li>btop-cuda</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>btop</summary>
<pre>source root is source
Running phase: patchPhase
Running phase: configurePhase
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/share/doc/btop -DCMAKE_INSTALL_INFODIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/include -DCMAKE_INSTALL_SBINDIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/cpjrfk28hkjl7qxxamsc0v9sndj41wc1-clang-wrapper-19.1.7/bin/strip -DCMAKE_RANLIB=/nix/store/cpjrfk28hkjl7qxxamsc0v9sndj41wc1-clang-wrapper-19.1.7/bin/ranlib -DCMAKE_AR=/nix/store/cpjrfk28hkjl7qxxamsc0v9sndj41wc1-clang-wrapper-19.1.7/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/9ldh0hlvh4gi4m62cz1dfjk6q9fimb63-btop-1.4.4
-- The CXX compiler identification is Clang 19.1.7
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/cpjrfk28hkjl7qxxamsc0v9sndj41wc1-clang-wrapper-19.1.7/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /nix/store/0ywb0vafxr8glh4baja4b44cz8ag8zkr-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:68 (message):
  IPO is not supported (check failed to compile).
Call Stack (most recent call first):
  /nix/store/0ywb0vafxr8glh4baja4b44cz8ag8zkr-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:153 (_ipo_not_supported)
  /nix/store/0ywb0vafxr8glh4baja4b44cz8ag8zkr-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:261 (_ipo_run_language_check)
  CMakeLists.txt:67 (check_ipo_supported)


-- Configuring incomplete, errors occurred!</pre>
</details>
<details>
<summary>btop-cuda</summary>
<pre>source root is source
Running phase: patchPhase
Running phase: configurePhase
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/share/doc/btop -DCMAKE_INSTALL_INFODIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/include -DCMAKE_INSTALL_SBINDIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/cpjrfk28hkjl7qxxamsc0v9sndj41wc1-clang-wrapper-19.1.7/bin/strip -DCMAKE_RANLIB=/nix/store/cpjrfk28hkjl7qxxamsc0v9sndj41wc1-clang-wrapper-19.1.7/bin/ranlib -DCMAKE_AR=/nix/store/cpjrfk28hkjl7qxxamsc0v9sndj41wc1-clang-wrapper-19.1.7/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/mvwv64cyj38y98m7n6plhf9cd5l65adr-btop-1.4.4
-- The CXX compiler identification is Clang 19.1.7
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/cpjrfk28hkjl7qxxamsc0v9sndj41wc1-clang-wrapper-19.1.7/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /nix/store/0ywb0vafxr8glh4baja4b44cz8ag8zkr-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:68 (message):
  IPO is not supported (check failed to compile).
Call Stack (most recent call first):
  /nix/store/0ywb0vafxr8glh4baja4b44cz8ag8zkr-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:153 (_ipo_not_supported)
  /nix/store/0ywb0vafxr8glh4baja4b44cz8ag8zkr-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:261 (_ipo_run_language_check)
  CMakeLists.txt:67 (check_ipo_supported)


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>btop</summary>
<pre>@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/share/doc/btop -DCMAKE_INSTALL_INFODIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/include -DCMAKE_INSTALL_SBINDIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/1g645wydyvns7dy8gy27dq0n3rbkd712-clang-wrapper-19.1.7/bin/strip -DCMAKE_RANLIB=/nix/store/1g645wydyvns7dy8gy27dq0n3rbkd712-clang-wrapper-19.1.7/bin/ranlib -DCMAKE_AR=/nix/store/1g645wydyvns7dy8gy27dq0n3rbkd712-clang-wrapper-19.1.7/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/hbl31b5lzkvwh3kh6k448af05ba7gmr0-btop-1.4.4
-- The CXX compiler identification is Clang 19.1.7
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/1g645wydyvns7dy8gy27dq0n3rbkd712-clang-wrapper-19.1.7/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /nix/store/6m8dh17b5ylql37wnkwjcag45i8nk2by-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:68 (message):
  IPO is not supported (check failed to compile).
Call Stack (most recent call first):
  /nix/store/6m8dh17b5ylql37wnkwjcag45i8nk2by-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:153 (_ipo_not_supported)
  /nix/store/6m8dh17b5ylql37wnkwjcag45i8nk2by-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:261 (_ipo_run_language_check)
  CMakeLists.txt:67 (check_ipo_supported)


-- Configuring incomplete, errors occurred!</pre>
</details>
<details>
<summary>btop-cuda</summary>
<pre>@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/share/doc/btop -DCMAKE_INSTALL_INFODIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/include -DCMAKE_INSTALL_SBINDIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/1g645wydyvns7dy8gy27dq0n3rbkd712-clang-wrapper-19.1.7/bin/strip -DCMAKE_RANLIB=/nix/store/1g645wydyvns7dy8gy27dq0n3rbkd712-clang-wrapper-19.1.7/bin/ranlib -DCMAKE_AR=/nix/store/1g645wydyvns7dy8gy27dq0n3rbkd712-clang-wrapper-19.1.7/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/a0hmlynfgxgshg13bwbvjvphh6x9cagp-btop-1.4.4
-- The CXX compiler identification is Clang 19.1.7
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/1g645wydyvns7dy8gy27dq0n3rbkd712-clang-wrapper-19.1.7/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /nix/store/6m8dh17b5ylql37wnkwjcag45i8nk2by-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:68 (message):
  IPO is not supported (check failed to compile).
Call Stack (most recent call first):
  /nix/store/6m8dh17b5ylql37wnkwjcag45i8nk2by-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:153 (_ipo_not_supported)
  /nix/store/6m8dh17b5ylql37wnkwjcag45i8nk2by-cmake-3.31.7/share/cmake-3.31/Modules/CheckIPOSupported.cmake:261 (_ipo_run_language_check)
  CMakeLists.txt:67 (check_ipo_supported)


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

--- a/pkgs/by-name/bt/btop/package.nix
+++ b/pkgs/by-name/bt/btop/package.nix
@@ -15,13 +15,13 @@
 
 stdenv.mkDerivation rec {
   pname = "btop";
-  version = "1.4.3";
+  version = "1.4.4";
 
   src = fetchFromGitHub {
     owner = "aristocratos";
     repo = "btop";
     rev = "v${version}";
-    hash = "sha256-4x2vGmH2dfHZHG+zj2KGsL/pRNIZ8K8sXYRHy0io5IE=";
+    hash = "sha256-4H9UjewJ7UFQtTQYwvHZL3ecPiChpfT6LEZwbdBCIa0=";
   };
 
   nativeBuildInputs =
@@ -37,6 +37,11 @@ stdenv.mkDerivation rec {
   ];
 
   installFlags = [ "PREFIX=$(out)" ];
+
+  # fix build on darwin (see https://github.com/NixOS/nixpkgs/pull/422218#issuecomment-3039181870 and https://github.com/aristocratos/btop/pull/1173)
+  cmakeFlags = [
+    (lib.cmakeBool "BTOP_LTO" (!stdenv.hostPlatform.isDarwin))
+  ];
 
   postInstall = ''
     ${removeReferencesTo}/bin/remove-references-to -t ${stdenv.cc.cc} $(readlink -f $out/bin/btop)

